src/Test/QuickCheck/Classes/Generic.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+#if MIN_VERSION_base(4,12,0)
+{-# LANGUAGE QuantifiedConstraints #-}
+#endif
+{-# OPTIONS_GHC -Wall #-}
+
+module Test.QuickCheck.Classes.Generic
+  (
+#if MIN_VERSION_base(4,6,0)
+    genericLaws
+  , genericLawsU
+#endif
+#if MIN_VERSION_base(4,12,0) 
+--  , generic1Laws
+#endif
+  ) where
+
+#if MIN_VERSION_base(4,6,0)
+import Control.Applicative
+import Data.Semigroup
+import Data.Monoid
+import GHC.Generics
+#if MIN_VERSION_base(4,9,0)
+import Data.Functor.Classes
+#endif
+import Data.Proxy (Proxy(Proxy))
+import Test.QuickCheck
+import Test.QuickCheck.Property (Property)
+
+import Test.QuickCheck.Classes.Common (Laws(..), Apply(..))
+
+-- | Tests the following properties:
+--
+-- [/Conjunction Idempotence/]
```

Comment:
These laws don't appear to correspond to those of `Generic`.

Reply:
Right, artifact of some copy-pasting. Will fix.